pkgs/development/libraries/librsb/default.nix

```
@@ -0,0 +1,86 @@
+{ lib, stdenv, fetchurl
+, gfortran
+, pkg-config, libtool
+, m4, gnum4
+, file
+# Memory Hierarchy (End-user can provide this.)
+, memHierarchy ? ""
+# Headers/Libraries
+, blas, zlib
+# RPC headers (rpc/xdr.h)
+, openmpi, ntirpc
+, help2man
+, doxygen
+, octave
+}:
+
+stdenv.mkDerivation rec {
+  pname = "librsb";
+  version = "1.2.0.9";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/${pname}/${pname}-${version}.tar.gz";
+    sha256 = "1ynrsgnvv1jfm8dv3jwjrip9x9icxv7w3qrk149025j6fbaza8gl";
+  };
+
+  # The default configure flags are still present when building
+  # --disable-static --disable-dependency-tracking
+  # Along with the --prefix=... flag (but we want that one).
+  configureFlags = [
+    "--enable-static"
+    "--enable-doc-build"
+    "--enable-octave-testing"
+    "--enable-sparse-blas-interface"
+    "--enable-fortran-module-install"
+    "--enable-pkg-config-install"
+    "--enable-matrix-types=all"
+  ]
+  ++ lib.optionals (zlib != null) [ "--with-zlib=${zlib}/lib/libz.so" ]
+  ++ lib.optionals (memHierarchy != "") [ "--with-memhinfo=${memHierarchy}" ];
```

Is there a use case when you would want to overwrite them? If not we can simplify this.

```suggestion
    "--enable-matrix-types=all"
    "--with-zlib=${zlib}/lib/libz.so"
    "--with-memhinfo=${memHierarchy}"
  ];
```

---

For `memHierarchy`, yes. If you want to tailor the build more specifically for *your* computer's cache layout, then you will definitely want to override it. In `all-packages.nix`, there is a default parameter argument of 
```nix
memHierarchy = "L3:16/64/8192K,L2:16/64/2048K,L1:8/64/16K";
```
This set of memory layout compilation configurations were taken from OpenSuSe's build systems. That configuration is good enough for most regular users of this library. But, if a person *really* wants to fine-tune it to their CPU, I left `memHierarchy` like that to be overwritten.

`zlib` is like that only because it's an optional dependency. The library/program can run without it just fine, it just can't be configured to use the library.

---

Most packaged in nixpkgs are build with all features of have a full and minimal variant. I would prefer if we would just always include zlib to keep it simple.

---

I'm perfectly find with that. How about the `memHierarchy` one though? I feel that should be something very easy for the end-user to override. I'm just not sure how to best go about that.

---

``librsb.override{memHierarchy = ""}`` ?